homeassistant/components/hp_ilo/sensor.py

```diff
@@ -120,18 +131,31 @@ def __init__(
         sensor_name,
         sensor_value_template,
         unit_of_measurement,
+        device_class,
+        state_class,
     ):
         """Initialize the HP iLO sensor."""
         self._hass = hass
         self._name = sensor_name
-        self._unit_of_measurement = unit_of_measurement
         self._ilo_function = SENSOR_TYPES[sensor_type][1]
         self.hp_ilo_data = hp_ilo_data
 
         if sensor_value_template is not None:
             sensor_value_template.hass = hass
         self._sensor_value_template = sensor_value_template
 
+        if sensor_type == "server_power_readings":
+            if unit_of_measurement is None:
+                unit_of_measurement = "W"
+            if device_class is None:
+                device_class = "power"
+            if state_class is None:
+                state_class = "measurement"
+
+        self._unit_of_measurement = unit_of_measurement
+        self._device_class = device_class
+        self._state_class = state_class
```

Comment:
Use shorthand attributes instead
```suggestion
        self._attr_device_class = device_class
        self._attr_state_class = state_class
```